### PR TITLE
Add `start_on`

### PR DESCRIPTION
**Pull request by elbeno**
This PR is a follow-on from #54

### CODE REVIEW

**Comment by ericniebler** [comment]
Regarding the renames, LEWG hasn't seen those papers yet.

**Comment by elbeno** [comment]
Renames are easy; I'm trying to get up to date with nomenclature (even pre-approved) before Aspen so that my talk isn't too confusing!

**Comment by ericniebler** [comment]
that makes sense, i do find the new names less confusing

File: docs/sender_adaptors.adoc [hunk 1]
```diff
@@ -197,6 +176,25 @@ multiple times.
 `split` turns a single shot sender into a multishot sender. It has no effect
 when called on a multishot sender.
 
+=== `start_on`
+
+`start_on` takes a scheduler and a sender, and composes them so that the work
+will run on that scheduler. It chains the sender work on to the result of
```
Comment (mjcaisse-intel) [inline docs/sender_adaptors.adoc:182]: onto

File: docs/sender_adaptors.adoc [hunk 2]
```diff
@@ -197,6 +176,25 @@ multiple times.
...
+calling `schedule`.
+
+[source,cpp]
+----
+// s is a scheduler
+auto sndr = async::start_on(s, async::just(42));
+// when run, sndr will execute on the compute resource specified by s, producing 42
+----
+
+`start_on` is equivalent to `seq(scheduler.schedule(), sender`:
```
Comment (mjcaisse-intel) [inline docs/sender_adaptors.adoc:192]: missing ending `)`

**Review by mjcaisse-intel — APPROVED** [review]
Documentation comments.

File: docs/sender_adaptors.adoc (hunk 2 above)
Comment (elbeno) [reply docs/sender_adaptors.adoc]: good spot

**Pull request merged**